### PR TITLE
blocked-edges/4.7.4*: Targeted edge blocking and version 1.1.0

### PR DESCRIPTION
**Pull request by wking**
So we can explain why we're blocking the different edges: openshift/enhancements#821.

The `zz` in the filename for the vSphere hostname block ensures that one sorts last alphabetically, because it's the broadest block, and legacy Cincinnati services will prefer the final regular expression they load for a given `to` target.

### CODE REVIEW

**Comment by wking** [comment]
/hold

Don't want to rush this one in ;)

**Comment by wking** [comment]
Checking a66bd51 on a 4.6.0 OSUS installed on 4.8.11 OpenShift (via #1057 pointed at this branch):

```console
$ curl -ksH accept:application/json 'https://example-policy-engine-route-openshift-update-service.apps.ci-ln-l4qnyb2-f76d1.origin-ci-int-gce.dev.openshift.com/api/upgrades_info/v1/graph?channel=candidate-4.7' | jq -r '.nodes[] | select(.version == "4.7.4").metadata["io.openshift.upgrades.graph.previous.remove_regex"]'
.*
$ curl -ksH accept:application/json 'https://example-policy-engine-route-openshift-update-service.apps.ci-ln-l4qnyb2-f76d1.origin-ci-int-gce.dev.openshift.com/api/upgrades_info/v1/graph?channel=candidate-4.7' | jq -r '(.nodes | with_entries(.key |= tostring)) as $nodes_by_index | .edges[] | $nodes_by_index[.[0] | tostring].version + " -> " + $nodes_by_index[.[1] | tostring].version' | grep ' -> 4.7.4'
...no hits...
```

So looks good to me vs. an old Cincinnati.

File: README.md
Comment (vrutkovs) [inline README.md:106]: IMO justification link should not be optional

File: README.md
Comment (vrutkovs) [inline README.md:108]: Do we need to come up with a name when its tied to a bug is (almost) all cases?

File: README.md
Comment (wking) [reply README.md:106]: I think it should, and there's a section in the in-flight enhancement explaining my motivation, but let's hash this out [in the relevant enhancements thread][1].

[1]: https://github.com/openshift/enhancements/pull/821#discussion_r706179172

File: README.md
Comment (wking) [reply README.md:108]: This slug [eventually feeds the `Recommended` condition's `reason` property][1].  I think we want that, although see the *Metadata to include when blocking a conditional request* section in the enhancement for discussion.

[1]: https://github.com/openshift/enhancements/pull/821#discussion_r705538121

**Review by vrutkovs — APPROVED** [review]
/lgtm

File: README.md
Comment (LalatenduMohanty) [reply README.md:108]: I wonder if we maintain the list of name i.e. `ClusterOperatorStatusCondition` `reason` it might be useful to create some kind of glossary

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Comment by wking** [comment]
Checking now that this is live.  From shortly before the PR merged:

```console
$ curl -s 'https://api.openshift.com/api/upgrades_info/v1/graph?channel=fast-4.7' >fast-4.7-pre.json
$ curl -s 'https://api.openshift.com/api/upgrades_info/v1/graph?channel=fast-4.8' >fast-4.8-pre.json
```

And from after the PR merged:

```console
$ curl -s 'https://api.openshift.com/api/upgrades_info/v1/graph?channel=fast-4.7' >fast-4.7-post.json
$ curl -s 'https://api.openshift.com/api/upgrades_info/v1/graph?channel=fast-4.8' >fast-4.8-post.json
```

Writing a comparer:

```console
$ edges() {
>   jq -r '(.nodes | with_entries(.key |= tostring)) as $nodes_by_index | .edges[] | $nodes_by_index[(.[0] | tostring)].version + " -> " + $nodes_by_index[(.[1] | tostring)].version' "${1}" | sort -V
> }
```

And using it:

```console
$ diff -u <(edges fast-4.7-pre.json) <(edges fast-4.7-post.json) && echo 'no changes'
no changes
$ diff -u <(edges fast-4.8-pre.json) <(edges fast-4.8-post.json) && echo 'no changes'
no changes
```

So hooray, we're good vs. modern Cincinnati too.

**Pull request merged**
